### PR TITLE
Make errors actionable when failing to deserialize a ChannelManager

### PR DESCRIPTION
**Pull request by TheBlueMatt**
See https://github.com/rust-bitcoin/rust-lightning/pull/965#discussion_r661043915

### CODE REVIEW

File: lightning/src/ln/channelmanager.rs [hunk 1]
```diff
@@ -4681,6 +4683,8 @@ impl<'a, Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 				log_error!(args.logger, "Missing ChannelMonitor for channel {} needed by ChannelManager.", log_bytes!(channel.channel_id()));
 				log_error!(args.logger, " The chain::Watch API *requires* that monitors are persisted durably before returning,");
 				log_error!(args.logger, " client applications must ensure that ChannelMonitor data is always available and the latest to avoid funds loss!");
+				log_error!(args.logger, " Without the ChannelMonitor we cannot continue without risking funds.");
+				log_error!(args.logger, " Please ensure the chain::Watch API requirements are met file a bug report at https://github.com/rust-bitcoin/rust-lightning");
```
Comment (ariard) [inline lightning/src/ln/channelmanager.rs:4687]: Just need a "and" between "Please ensure the chain::Watch API requirements are met and file a bug report at https://github.com/rust-bitcoin/rust-lightning" ?

**Review by ariard — COMMENTED** [review]
Otherwise SGTM

**Review by valentinewallace — APPROVED** [review]

**Comment by TheBlueMatt** [comment]
Squashed without diff:

```
$ git diff-tree -U1 b918220d e8110ab3
$
```

**Comment by ariard** [comment]
ACK e8110ab33

For the records about e8110ab, it has been there since #91 : https://github.com/rust-bitcoin/rust-lightning/pull/91/files#diff-8b7223022de62dd3acad8a134eb6fc1cb291b0144df8fb0bb63c454860471a77R167

**Comment by TheBlueMatt** [comment]
> For the records about e8110ab, it has been there since #91 : https://github.com/rust-bitcoin/rust-lightning/pull/91/files#diff-8b7223022de62dd3acad8a134eb6fc1cb291b0144df8fb0bb63c454860471a77R167

That specific code yes, but I naively swapped the sort order of log levels in #965, which caused the regression.

**Pull request merged**